library/test/src/lib.rs

```diff
@@ -781,11 +781,21 @@ fn run_test_in_spawned_subprocess(
             builtin_panic_hook(info);
         }
 
-        if let TrOk = test_result {
-            process::exit(test_result::TR_OK);
+        let return_code = if test_result == TrOk {
+            test_result::TR_OK
         } else {
-            process::exit(test_result::TR_FAILED);
+            test_result::TR_FAILED
+        };
+
+        // Skip atexit hooks if the test panicked. This will prevent (among
+        // other things) leak sanitizer from being run, which should not have
+        // its atexit hook run when a test panics. This simulates the behavior
+        // of aborting more closely on unix.
```

Comment:
If we're trying to simulate an abort (not exactly sure that we are, maybe this function only runs in a panic=abort context?) then perhaps we should be aborting directly? Or do we need a specific exit code for some reason?

Reply:
The exit code is used to distinguish successful and failed tests. If a `#[should_panic]` test panics, it should still return `TR_OK`. Merely doing a check if the process aborted and letting panics always abort would cause `#[should_panic]` tests that hard crash to be incorrectly marked as successful. It would also cause tests that use `std::process::exit(0)` to be marked as successful.